test/evaluate/ISO-Fortran-binding.cc

```diff
@@ -210,14 +210,459 @@ int main() {
+  CFI_index_t strides[rank] = {2, 0, 2};
+  constexpr CFI_rank_t resultRank = rank - 1;
+
+  CFI_CDESC_T(resultRank) resultStorage;
+  CFI_cdesc_t *result = reinterpret_cast<CFI_cdesc_t *>(&resultStorage);
+  retCode = CFI_establish(result, nullptr, CFI_attribute_other,
+      CFI_type_long_long, 0, resultRank, nullptr);
+  testPreConditions &= (retCode == CFI_SUCCESS);
+
+  if (!testPreConditions) {
+    MATCH(true, testPreConditions);
+    return;
+  }
+
+  retCode = CFI_section(
+      result, source, lb, ub, strides);  // Fortran B = A(2:4:2, 5:5:0, 4:6:2)
+  MATCH(true, retCode == CFI_SUCCESS);
+
+  const int lbs0 = source->dim[0].lower_bound;
+  const int lbs1 = source->dim[1].lower_bound;
+  const int lbs2 = source->dim[2].lower_bound;
+
+  CFI_index_t resJ{result->dim[1].lower_bound};
+  for (CFI_index_t k{lb[2]}; k <= ub[2]; k += strides[2]) {
+    for (CFI_index_t j{lb[1]}; j <= ub[1]; j += (strides[1] ? strides[1] : 1)) {
+      CFI_index_t resI{result->dim[0].lower_bound};
+      for (CFI_index_t i{lb[0]}; i <= ub[0]; i += strides[0]) {
+        // check A(i,j,k) == B(resI, resJ) == array[k-1][j-1][i-1]
+        const CFI_index_t resSubcripts[] = {resI, resJ};
+        const CFI_index_t srcSubcripts[] = {i, j, k};
+        MATCH(true,
+            CFI_address(source, srcSubcripts) ==
+                CFI_address(result, resSubcripts));
+        MATCH(true,
+            CFI_address(source, srcSubcripts) ==
+                &array[k - lbs2][j - lbs1][i - lbs0]);
+        ++resI;
+      }
+    }
+    ++resJ;
+  }
+
+  strides[0] = -1;
+  lb[0] = 4;
+  ub[0] = 2;
+  retCode = CFI_section(
+      result, source, lb, ub, strides);  // Fortran B = A(4:2:-1, 5:5:0, 4:6:2)
+  MATCH(true, retCode == CFI_SUCCESS);
+
+  resJ = result->dim[1].lower_bound;
+  for (CFI_index_t k{lb[2]}; k <= ub[2]; k += strides[2]) {
+    for (CFI_index_t j{lb[1]}; j <= ub[1]; j += 1) {
+      CFI_index_t resI{result->dim[1].lower_bound + result->dim[0].extent - 1};
+      for (CFI_index_t i{2}; i <= 4; ++i) {
+        // check A(i,j,k) == B(resI, resJ) == array[k-1][j-1][i-1]
+        const CFI_index_t resSubcripts[] = {resI, resJ};
+        const CFI_index_t srcSubcripts[] = {i, j, k};
+        MATCH(true,
+            CFI_address(source, srcSubcripts) ==
+                CFI_address(result, resSubcripts));
+        MATCH(true,
+            CFI_address(source, srcSubcripts) ==
+                &array[k - lbs2][j - lbs1][i - lbs0]);
+        --resI;
+      }
+    }
+    ++resJ;
+  }
+}
+
+static void run_CFI_select_part_tests() {
+  constexpr std::size_t name_len = 5;
+  typedef struct {
+    double distance;
+    int stars;
+    char name[name_len];
+  } Galaxy;
+
+  const CFI_rank_t rank{2};
+  constexpr CFI_index_t universeSize[] = {2, 3};
+  Galaxy universe[universeSize[1]][universeSize[0]];
+
+  for (int j{0}; j < universeSize[1]; ++j) {
+    for (int i{0}; i < universeSize[0]; ++i) {
+      universe[j][i].distance = i + j * 32;
```

Comment:
This looks like a test, but can't these loops be interchanged, etc. to initialize the buffers in C's row-major order?

Reply:
It is because I wanted it to be clear that this code is initializing data related to a Fortran variable. I changed it to the usual C's row major order and I have added a comment regarding what Fortran indexes it initializes.